encodings/protobuf/README.md

```diff
@@ -0,0 +1,225 @@
+separate control file (the canonical version of which is included in
+this repo in `control.json`) is used "remember" the field ids assigned
+to fields in messages.
+
+### Enumerated Values
+
+Enumerated **integer** values are built using protobuf's `enum`
+declaration.  Because OCSF often (not always) refines the set of
+enumerands on a per-event basis, the protobuf encoding supports that
+by making the enumeration types sub-names of the message types.
+Furthermore, because Protobuf uses "C++" scoping rules, in which
+enumerated values are in the same scope as the enum type itself
+(rather than children of the type), we mangle the enumerand names as
+well.
+
+For example:
+```
+message FileActivity {
+    enum ActionId {
+        UNKNOWN_ActionId = 0;  // The action was unknown. The <code>disposition_id</code>
+                               // attribute may still be set to a non-unknown value, for
+                               // example 'Count', 'Uncorrected', 'Isolated',
+                               // 'Quarantined' or 'Exonerated'.
+        ALLOWED_ActionId = 1;  // The activity was allowed. The
+                               // <code>disposition_id</code> attribute should be set to a
+                               // value that conforms to this action, for example
+                               // 'Allowed', 'Approved', 'Delayed', 'No Action', 'Count'
+                               // etc.
+        DENIED_ActionId  = 2;  // The attempted activity was denied. The
+                               // <code>disposition_id</code> attribute should be set to a
+                               // value that conforms to this action, for example
+                               // 'Blocked', 'Rejected', 'Quarantined', 'Isolated',
+                               // 'Dropped', 'Access Revoked, etc.
+        OTHER_ActionId   = 99; // The action was not mapped. See the <code>action</code>
+                               // attribute, which contains a data source specific value.
+    }
+    enum ActivityId {
+        UNKNOWN_ActivityId        = 0; 
+        CREATE_ActivityId         = 1;  // A request to create a new file on a file
+                                        // system.
+        READ_ActivityId           = 2;  // A request to read data from a file on a file
+                                        // system.
+        ...
+    }
+    ...
+    ActionId action_id = 3;
+    ActivityId activity_id = 4;
+    ...
+}
+```
+
+The protobuf encoding preserves the numerical value of the enumerand.
+This is mostly for consistency with OCSF, but also helps with JSON
+Encoding interoperability (see below).
+
+However, not every enumeration in OCSF includes a 0 value, which is
+required in proto3, so in cases where the 0 value is missing, an
+`UNKNOWN` enumerand with the 0 value is inserted.
+
+No special treatment is given for enumerated **string** values in
+OCSF, they are just treated as strings by the protobuf encoding.
```

Comment:
Indeed, the existing string enumeration (there's only one at the moment AFAIK) does _not_ have "unknown" and "other" enumeration values. Even for integer enumerations, there's no hard reason for these to include the `0` unknown and `99` other values. For example an enumeration of day of week using 0 to 6 or 1 to 7 might not need values for unknown and certainly not for other.

(There's no changed needed here, though at your discretion, you could expand on this more. However expanding more probably doesn't help explain what's going on, so could just add noise here.)